driver/main.c

```diff
@@ -2189,35 +2232,35 @@ TRACEPOINT_PROBE(syscall_enter_probe, struct pt_regs *regs, long id)
 	g_n_tracepoint_hit_inc();
 
 	table_index = id - SYSCALL_TABLE_ID0;
-	if (likely(table_index >= 0 && table_index < SYSCALL_TABLE_SIZE)) {
-		struct event_data_t event_data;
-		int used = cur_g_syscall_table[table_index].flags & UF_USED;
-		enum syscall_flags drop_flags = cur_g_syscall_table[table_index].flags;
-		ppm_event_code type;
+	if (unlikely(table_index < 0 || table_index >= SYSCALL_TABLE_SIZE))
+	{
+		return;
+	}
 
 #ifdef _HAS_SOCKETCALL
-		if (id == socketcall_syscall) {
-			used = true;
-			drop_flags = UF_NEVER_DROP;
-			type = PPME_GENERIC_E;
-		} else
-			type = cur_g_syscall_table[table_index].enter_event_type;
-#else
-		type = cur_g_syscall_table[table_index].enter_event_type;
-#endif
-
-		event_data.category = PPMC_SYSCALL;
-		event_data.event_info.syscall_data.regs = regs;
-		event_data.event_info.syscall_data.id = id;
-		event_data.event_info.syscall_data.cur_g_syscall_table = cur_g_syscall_table;
-		event_data.socketcall_syscall = socketcall_syscall;
-		event_data.compat = compat;
-
-		if (used)
-			record_event_all_consumers(type, drop_flags, &event_data, KMOD_PROG_SYS_ENTER);
-		else
-			record_event_all_consumers(PPME_GENERIC_E, UF_ALWAYS_DROP, &event_data, KMOD_PROG_SYS_ENTER);
+	if (id == socketcall_syscall)
+	{
+		id = convert_network_syscalls(regs);
+		table_index = id - SYSCALL_TABLE_ID0;
+		is_socketcall = true;
 	}
+#endif
+	struct event_data_t event_data;
+	int used = cur_g_syscall_table[table_index].flags & UF_USED;
+	enum syscall_flags drop_flags = cur_g_syscall_table[table_index].flags;
+	ppm_event_code type = cur_g_syscall_table[table_index].enter_event_type;
```

Comment:
we cannot declare them here with ISO C90 we need to put them at the beginning of the filler, in this way we can avoid also some local variables like

```c
	const struct syscall_evt_pair *cur_g_syscall_table = g_syscall_table;
	bool compat = false;
	int socketcall_syscall = __NR_socketcall;
```

Reply:
It was already like this before :/

Reply:
Oh right, forgot that!

Reply:
it was inside an `if`

Reply:
> in this way we can avoid also some local variables like

I think they look ok there btw; i'd avoid accessing `event_data` on multiple lines; much better to gather all the info and then fill it in a single place so that it has more visibility.